src/components/Checkbox.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @typedef CheckboxBaseProps
+ * @prop {string} name - The `name` of the checkbox.
+ * @prop {import('preact').Ref<HTMLInputElement>} [inputRef]
+ * @prop {(checked: boolean) => void} [onToggle] - Callback when checkbox is
+ *   checked/unchecked
+ * @prop {never} [type] - Type is always 'checkbox'
+ * @prop {never} [children] - Children are not allowed
+ *
+ * @typedef {Omit<import('Preact').JSX.HTMLAttributes<HTMLInputElement>, 'onToggle'> & CheckboxBaseProps} CheckboxProps
+ *
+ * @typedef LabeledCheckboxProps
+ * @prop {string} label - Label text
```

Comment:
I have a sense we might run into a need to have a label that is richer than a string in the future (e.g. contains a link, or an image, or something). We can probably defer dealing with this for now.